pkg/operator/ceph/object/objectstore.go

```diff
@@ -954,40 +957,109 @@ func zoneUpdateWorkaround(objContext *Context, expectedZone, gotZone map[string]
 	if err != nil {
 		return err
 	}
+
+	// check expected number of placements.
+	// If number is different, try to check if 'default-placement' was not-removed and remove it manually.
+	// Otherwise return error.
+	// see https://tracker.ceph.com/issues/68775
+	placementWasNotRemoved := false
 	if len(expected) != len(got) {
-		// should not happen
-		return fmt.Errorf("placements were not applied to zone config: expected %+v, got %+v", expected, got)
+		if len(expected) == len(got)-1 {
+			// note that there are extra placement to handle it later:
+			placementWasNotRemoved = true
+		} else {
+			// should not happen
+			return fmt.Errorf("placements were not applied to zone config: expected %+v, got %+v", expected, got)
+		}
 	}
 
+	// flags to check if extra placement is named 'default-placement'
+	expHasDefault, gotHasDefault := false, false
 	// update pool placements one-by-one if needed
-	for i, expPl := range expected {
-		expPoolObj, ok := expPl.(map[string]interface{})
+	for i := range got {
```

Comment:
Could we rename the "got" var to something more descriptive? perhaps this? In general, I'd suggest "existing" in place of "got" for naming. 
```suggestion
	for i := range existingPlacementPools {
```

Reply:
done